### PR TITLE
Feature/aatf ae details report

### PR DESCRIPTION
**Pull request by willdot**
PBI: 73690

### CODE REVIEW

File: src/EA.Weee.DataAccess/StoredProcedure/StoredProcedures.cs [hunk 1]
```diff
@@ -585,5 +585,20 @@ public async Task<List<AatfReuseSitesData>> GetAllAatfReuseSitesCsvData(int comp
                     complianceYearParameter, authorityParameter, panAreaParameter)
                 .ToListAsync();
         }
+
+        public async Task<List<AatfAeDetailsData>> GetAatfAeDetailsCsvData(int complianceYear, int facilityType, Guid? authority, Guid? area, Guid? panArea)
+        {
+            SqlParameter complianceYearParameter = new SqlParameter("@ComplianceYear", complianceYear);
+            SqlParameter facilityTypeParameter = new SqlParameter("@FacilityType", facilityType);
+            SqlParameter authorityParameter = new SqlParameter("@CA", (object)authority ?? DBNull.Value);
+            SqlParameter panAreaParameter = new SqlParameter("@PanArea", (object)panArea ?? DBNull.Value);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.DataAccess/StoredProcedure/StoredProcedures.cs]: Can you add some integration tests, something similar to GetAatfSubmissionsTests

File: src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs [hunk 2]
```diff
@@ -127,6 +127,9 @@ public ActionResult ChooseReport(ChooseReportViewModel model)
                 case Reports.AatfReuseSitesData:
                     return RedirectToAction("AatfReuseSites", "AatfReports");
 
+                case Reports.AatfAeDetails:
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs]: Add test for this

File: src/EA.Weee.Database/scripts/Update/Release 2/sprint16/20190823-1438-create-sp-getAatfAeDetailsCsvData.sql [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		CASE WHEN o.Name IS NULL THEN o.TradingName ELSE o.Name END,
+		CONCAT(oa.Address1, COALESCE(', ' + NULLIF(oa.Address2, ''), ''), ', ', oa.TownOrCity,COALESCE(', ' + NULLIF(oa.CountyOrRegion, ''), ''), ''),
+		oa.Postcode
+	
+	FROM
+		AATF.AATF a 
+		JOIN Organisation.Organisation o ON a.OrganisationId  = o.Id
+		JOIN Organisation.Address oa ON o.BusinessAddressId = oa.Id
+		JOIN Lookup.CompetentAuthority ca ON a.CompetentAuthorityId = ca.Id
+		JOIN Lookup.PanArea pa on a.PanAreaId = pa.Id
```
Comment (SumeN) [inline src/EA.Weee.Database/scripts/Update/Release 2/sprint16/20190823-1438-create-sp-getAatfAeDetailsCsvData.sql]: The aatf's with CA not 'EA' will not have pan and La area. This has to be a left join and the table will have those columns as nullable

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs [hunk 4]
```diff
@@ -1167,5 +1167,154 @@ public async Task GetDownloadAatfReuseSitesCsv_ReturnsFileResultWithContentTypeO
             fileResult.FileContents.Should().Contain(new UTF8Encoding().GetBytes(csvData.FileContent));
             fileResult.FileDownloadName.Should().Be(CsvFilenameFormat.FormatFileName(csvData.FileName));
         }
+
+        [Fact]
+        public async Task GetAatfAeDetails_Always_ReturnsAatfAeDetailsuViewModel()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/AatfReportsControllerTests.cs]: Little typo :)

**Pull request merged**